### PR TITLE
hyperkit 0.20210107

### PR DESCRIPTION
**Pull request by Bo98**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by Bo98** [comment]
Testing a VM inside of a VM (the CI) is probably going to be a problem here.

**Comment by mitchblank** [comment]
Was there some progress on Big Sur that isn't yet reflected on https://github.com/moby/hyperkit/issues/297 ?

**Comment by Bo98** [comment]
Yes, https://github.com/moby/hyperkit/pull/307

**Comment by carlocab** [comment]
If we can't test this in a sandbox, we may as well just get rid of the test block (or put in some trivial test that will work in a sandbox).

The current bottles are failing tests anyway...

**Comment by mitchblank** [comment]
> If we can't test this in a sandbox, we may as well just get rid of the test block (or put in some trivial test that will work in a sandbox).

IMO the best option would be to leave the test but skip it on CI -- that way it at least can be run manually as part of formula maintenance.  However I haven't had luck in the past with getting changes like that past review: https://github.com/Homebrew/homebrew-core/pull/66485#issuecomment-740823983

**Comment by Bo98** [comment]
There's been some changes so the VM environment so let's see if anything's changed here.

File: Formula/hyperkit.rb [hunk 1]
```diff
@@ -24,23 +26,30 @@ class Hyperkit < Formula
...
+    system "opam", "pin", "add", "num", Formula["ocaml-num"].version.to_s, "--locked", "--fake"
+
+    # Allow ocamlfind to work within the opamroot.
+    ENV["OCAMLPATH"] = "#{opamroot}/default/lib:#{HOMEBREW_PREFIX}/lib/ocaml"
+    ENV["OCAMLFIND_DESTDIR"] = "#{opamroot}/default/lib"
+
+    system "opam", "exec", "--",
+           "opam", "install", "uri.4.1.0", "qcow.0.11.0", "conduit.2.1.0", "lwt.5.3.0",
+           "qcow-tool.0.11.0", "mirage-block-unix.2.12.0", "conf-libev.4-11", "logs.0.7.0", "fmt.0.8.9",
+           "mirage-unix.4.0.0", "prometheus-app.1.0"
```
Comment (carlocab) [inline Formula/hyperkit.rb:48]: Would be good to add a comment here about how to update these when needed.

**Comment by Bo98** [comment]
```
 # Error: /usr/local/lib/ocaml/findlib/topfind.cmi
#        is not a compiled interface for this version of OCaml.
```

Did we not revsion bump `ocaml-findlib` when updating OCaml?

**Comment by carlocab** [comment]
Unfortunately not: #71942

But CI didn't pick anything up with `ocaml-findlib`. `ocaml-findlib` was even used to build other formulae in the same PR.

**Comment by Bo98** [comment]
> But CI didn't pick anything up with `ocaml-findlib`. `ocaml-findlib` was even used to build other formulae in the same PR.

I suppose if it only uses the binary then there's no issues. There'll only be issues if it actually uses it as a library, and `opam` by default does not use system findlib (which the extra lines in this formula show).

**Comment by carlocab** [comment]
Oops. We should probably add `ocaml-findlib` to the comment at the start of the `ocaml` formula then.

**Comment by cho-m** [comment]
Tried another run with updated `ocaml-findlib`, but now back to failing on testcase again.

**Pull request closed without merging**